src/main/resources/templates/registration_page.html

```diff
@@ -25,18 +25,28 @@ <h1 class="mb-4 text-center">Links</h1>
     <form th:action="@{/register}" th:object="${user}" method="post" enctype="application/x-www-form-urlencoded" autocomplete="on">
         <div class="mb-3">
             <label for="username" class="form-label">Username</label>
-            <input type="text" class="form-control" id="username" aria-describedby="emailHelp" name="username" th:field="*{username}" value="{{old('username}}">
-            <div class="text-danger" th:if="${#fields.hasErrors('username')}" th:errors="*{username}">Name error</div>
+            <input type="text" class="form-control" th:unless="${param.hasErrors != null}" id="username" aria-describedby="username" name="username" th:field="*{username}">
```

Comment:
th:classappend !! 
Thanks, man!!!

Reply:
I don't get it.
Take a lood at Username div:
```
<div class="mb-3">
            <label for="username" class="form-label" >Username</label>
            <input type="text" class="form-control" th:errorclass="is-invalid" th:field="*{username}" id="username" aria-describedby="username" name="username">
            <div class="invalid-feedback" th:errors="*{username}">Name error</div>
            <div class="valid-feedback" th:unless="${#fields.hasErrors('username')}">Looks good!</div>
        </div>
```
It's incomplete but empty input should generate corresponding style. But it doesn't work so as error annotations of entity.

P.S.:  I've changed postMapping method this way 
```
if (result.hasErrors()) {
            return "redirect:/register";
        }
```
instead of "redirect:/register?hasErrors. I think it's not necessary and we can avoid it because Thymeleaf has everything we need. I just can't make it to work for now.

Reply:
Try this
```java
if (result.hasErrors()) {
     return "registration_page.html";
}

```

Reply:
Fixed. Let's move to next tasks...